api/v1beta2/well_known_attributes.go

```diff
@@ -0,0 +1,148 @@
+/*
+Copyright 2023 Red Hat, Inc.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package v1beta2
```

Comment:
Interesting. Thinking on defining this as part of the API even though it won't have direct dependency with the AuthConfig CRD?

Not saying it's wrong, just wondering... Selectors specified within an AuthConfig CR technically should include [JSON paths](https://github.com/Kuadrant/authorino/blob/main/docs/features.md#common-feature-json-paths-selector) that are valid within the possibilities of the Well-known attributes ([string modifiers](https://github.com/Kuadrant/authorino/blob/main/docs/features.md#string-modifiers) and [interpolation](https://github.com/Kuadrant/authorino/blob/main/docs/features.md#interpolation) aside.) This makes the Well-known attributes type useful for checking CRs, and therefore also an (indirect) constraint of the API maybe?

At the same time, the AuthConfig is an API that belongs to the control plane only, while the Well-known attributes struct cannot be instantiated other than in the data plane. It's only the skeleton of the type that matters for the API, but we won't ever have values for it at that stage. 🤔